src/components/molecules/ChatPoll/ChatPoll.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPoll } from "@fortawesome/free-solid-svg-icons";
+
+import { PollMessage, BaseMessageToDisplay, Vote } from "types/chat";
+
+import { WithId } from "utils/id";
+
+import { ChatMessageInfo } from "components/atoms/ChatMessageInfo";
+import { TextButton } from "components/atoms/TextButton";
+import Button from "components/atoms/Button";
+
+import { useUser } from "hooks/useUser";
+
+import "./ChatPoll.scss";
+
+export interface ChatPollProps {
+  pollData: WithId<BaseMessageToDisplay<PollMessage>>;
+  deletePoll: (pollId: string) => void;
+  voteInPoll: (votes: Vote[], pollId: string) => void;
+}
+
+export const ChatPoll: React.FC<ChatPollProps> = ({
+  pollData,
+  voteInPoll,
+  deletePoll,
+}) => {
+  const { user } = useUser();
+
+  const userId = user?.uid;
+
+  const { id, poll, votes, isMine } = pollData;
+  const { questions, topic, canBeDeleted } = poll;
+
+  const isVoted = userId
+    ? votes.map(({ userId }) => userId).includes(userId)
+    : false;
+
+  const isDeleted = isMine && canBeDeleted;
+
+  const containerStyles = classNames("ChatPoll", {
+    "ChatPoll--me": isMine,
+  });
+
+  const formattedQuestions = questions
+    .map((item) => ({
+      ...item,
+      count: votes.length
+        ? Math.floor(
+            (votes.filter(({ questionId }) => questionId === item.id).length /
+              votes.length) *
+              100
+          )
+        : 0,
+    }))
+    .sort((a, b) => b.count - a.count);
+
+  const handleVote = useCallback(
+    (question) => {
+      if (!userId) return;
+
+      const newVote = {
+        questionId: question.id,
+        userId,
+      };
+
+      voteInPoll([...votes, newVote], id);
+    },
+    [id, voteInPoll, votes, userId]
+  );
```

Comment:
you should just pass the userId & messageId & answerId. It doesn't need to contain all the other votes

Reply:
`voteInPoll` should also use named props, not positional params.

Reply:
fixed in 0ec9686b7958dc8253722ffea7c1b5d62c52602e